app/services/curator/digital_object_factory_service.rb

```diff
@@ -6,25 +6,17 @@ class DigitalObjectFactoryService < Services::Base
 
     # TODO: set relationships for is_issue_of values
     def call
-      admin_set_ark_id = @json_attrs.dig('admin_set', 'ark_id')
-      begin
-        Curator.digital_object_class.transaction do
-          admin_set = Curator.collection_class.find_by(ark_id: admin_set_ark_id)
-          raise "AdminSet #{admin_set_ark_id} not found!" unless admin_set
-
-          digital_object = Curator.digital_object_class.find_or_initialize_by(ark_id: @ark_id)
+      with_transaction do
+        admin_set_ark_id = @json_attrs.dig('admin_set', 'ark_id')
+        collection_ark_ids = @json_attrs.fetch('exemplary_image_of', []).pluck('ark_id')
```

Comment:
Can't believe I missed this earlier (and I introduced this error, I think) -- but this should be `@json_attrs.fetch('is_member_of_collection', []).pluck('ark_id')`.

This is likely the cause of an issue I noticed where if there is an invalid/non-existent `ark_id` in the `is_member_of_collection` hash, the DigitalObject is still created and a 201 response rendered.

Reply:
Fixed that in next set of commits. Also handled the case for if the collections are not found